pkg/arvo/lib/shoe.hoon

```diff
@@ -0,0 +1,341 @@
+::  shoe: console application library
+::
+::    /lib/sole: draw some characters
+::    /lib/shoe: draw the rest of the fscking app
+::
+::    call +agent with a type, then call the resulting function with a core
+::    of the shape described in +shoe.
+::    you may produce classic gall cards and "shoe-effects", shorthands for
+::    sending cli events to connected clients.
+::    default implementations for the shoe-specific arms are in +default.
+::    for a simple usage example, see /app/shoe.
+::
+/-  *sole
+/+  sole, auto=language-server-complete
+|%
++$  state-0
+  $:  %0
+      soles=(map @ta sole-share)
+  ==
+::  $card: standard gall cards plus shoe effects
+::
++$  card
+  $%  card:agent:gall
+      [%shoe sole-ids=(list @ta) effect=shoe-effect]  ::  ~ sends to all soles
+  ==
+::  $shoe-effect: easier sole-effects
+::
++$  shoe-effect
+  $%  [%sole effect=sole-effect]
+      ::TODO  complex screen-draw effects
```

Comment:
I like having this layer to put more complex screen-drawing before deciding to support it through the whole stack.